src/main/java/net/minecraftforge/client/model/ModelDynBucket.java

```diff
@@ -0,0 +1,243 @@
+
+    // Do the textures for interiors exist?
+
+    try {
+      ResourceLocation loc = interior;
+      loc = new ResourceLocation(loc.getResourceDomain(), "textures/" + loc.getResourcePath() + ".png");
+      Minecraft.getMinecraft().getResourceManager().getAllResources(loc);
+      loc = front;
+      loc = new ResourceLocation(loc.getResourceDomain(), "textures/" + loc.getResourcePath() + ".png");
+      Minecraft.getMinecraft().getResourceManager().getAllResources(loc);
+
+      // both found
+      builder.add(interior);
+      builder.add(front);
+      hasTextures = true;
+    } catch(IOException e) {
+      // either interior, front or both textures weren't found. we don't require them.
+      hasTextures = false;
+    }
+
+    return builder.build();
+  }
+
+  @Override
+  public IFlexibleBakedModel bake(IModelState state, VertexFormat format,
+                                  Function<ResourceLocation, TextureAtlasSprite> bakedTextureGetter) {
+    // bucket model
+    IFlexibleBakedModel bakedBucket = bucket.bake(bucket.getDefaultState(), format, bakedTextureGetter);
+    TRSRTransformation transform = state.apply(bucket);
+
+    return new BakedDynBucket(bakedBucket, transform, FluidRegistry.WATER, hasTextures);
+  }
+
+  @Override
+  public IModelState getDefaultState() {
+    return bucket.getDefaultState();
+  }
+
+  public enum LoaderDynBucket implements ICustomModelLoader {
+    instance;
+
+    @Override
+    public boolean accepts(ResourceLocation modelLocation) {
+      return modelLocation.getResourceDomain().equals("forge") && modelLocation.getResourcePath().equals("models/item/dynbucket");
+    }
+
+    @Override
+    public IModel loadModel(ResourceLocation modelLocation) throws IOException {
+      IModel bucketModel = ModelLoaderRegistry.getModel(new ResourceLocation("minecraft", "item/bucket"));
+
+      return new ModelDynBucket(bucketModel);
+    }
+
+    @Override
+    public void onResourceManagerReload(IResourceManager resourceManager) {
+      // no need to clear cache since we create a new  model instance
+    }
+  }
+
+  // the dynamic bucket is based on the empty bucket
+  protected static class BakedDynBucket implements IFlexibleBakedModel, ISmartItemModel, IPerspectiveAwareModel {
+
+    private final IFlexibleBakedModel vanillaBakedBucket;
+    private final Map<String, IFlexibleBakedModel> cache = Maps.newHashMap(); // contains all the baked models since they'll never change
+
+    private final TRSRTransformation transform;
+    private final List<BakedQuad> generalQuads;
+    private final boolean hasTextures;
+
+    public BakedDynBucket(IFlexibleBakedModel bakedBucket, TRSRTransformation transform, Fluid fluid, boolean hasTextures) {
+      this.vanillaBakedBucket = bakedBucket;
+      this.hasTextures = hasTextures;
+
+      TextureAtlasSprite interiorSprite = Minecraft.getMinecraft().getTextureMapBlocks().getAtlasSprite(interior.toString());
```

Comment:
You're passed the bakedTextureGetter in the bake method for a reason - use that, instead of this.